programs/balanced-dollar/src/instructions.rs

```diff
@@ -254,15 +251,20 @@ pub fn get_handle_call_message_accounts<'info>(
 
         let user_address = Pubkey::from_str(account_from_network_address(message.to)?.as_str())
             .map_err(|_| BalancedDollarError::NotAnAddress)?;
+        let user_token_address =
+            get_associated_token_address(&user_address, &ctx.accounts.state.bn_usd_token);
+
         Ok(ParamAccounts {
-            accounts: get_accounts(ctx, user_address)?,
+            accounts: get_accounts(ctx, user_address, user_token_address)?,
         })
     } else if method == CROSS_TRANSFER_REVERT {
         let message = decode_cross_transfer_revert(&data)?;
         let user_address =
             Pubkey::from_str(&message.account).map_err(|_| BalancedDollarError::NotAnAddress)?;
+        let user_token_address =
```

Comment:
@bishalbikram this seems to create token account of token account, in the case of rollback, we will be already getting token account from xcall, so this will cause issue, and yeah same with the asset manager as well

Reply:
for the rollback case, we can just transfer back to from adress, as this is already a token account?